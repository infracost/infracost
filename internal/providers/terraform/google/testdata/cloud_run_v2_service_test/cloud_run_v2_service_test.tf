provider "google" {
  credentials = "{\"type\":\"service_account\"}"
  project     = "my-project"
}

resource "google_cloud_run_v2_service" "throttling_enabled" {
  name     = "cloudrun-service-test-with-cpu-idle"
  location = "europe-west3"
  ingress  = "INGRESS_TRAFFIC_ALL"

  template {
    containers {
      image = "us-docker.pkg.dev/cloudrun/container/hello"
    }
  }
}

resource "google_cloud_run_v2_service" "throttling_disabled" {
  name     = "cloudrun-service-test-with-cpu-idle"
  location = "europe-west3"
  ingress  = "INGRESS_TRAFFIC_ALL"

  template {
    containers {
      image = "us-docker.pkg.dev/cloudrun/container/hello"
      resources {
        limits = {
          cpu    = "1"
          memory = "512Mi"
        }
        cpu_idle = false
      }
    }
    scaling {
      min_instance_count = 2
    }
  }
}

resource "google_cloud_run_v2_service" "throttling_enabled_no_usage" {
  name     = "cloudrun-service-test-with-cpu-idle"
  location = "europe-west3"
  ingress  = "INGRESS_TRAFFIC_ALL"

  template {
    containers {
      image = "us-docker.pkg.dev/cloudrun/container/hello"
    }
  }
}
