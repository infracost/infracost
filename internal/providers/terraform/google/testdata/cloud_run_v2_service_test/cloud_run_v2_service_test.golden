
 Name                                                            Monthly Qty  Unit                      Monthly Cost   
                                                                                                                       
 google_cloud_run_v2_service.throttling_disabled                                                                       
 ├─ CPU allocation time (tier 2)                                   5,256,000  vCPU-seconds                   $113.53   
 └─ Memory allocation time (tier 2)                                2,628,000  GiB-seconds                      $6.31   
                                                                                                                       
 google_cloud_run_v2_service.throttling_enabled                                                                        
 ├─ CPU allocation time (tier 2)                                   1,800,000  vCPU-seconds                    $60.48   
 ├─ Memory allocation time (tier 2)                                  900,000  GiB-seconds                      $3.15   
 └─ Number of requests                                               500,000  requests                         $0.20   
                                                                                                                       
 google_cloud_run_v2_service.throttling_enabled_no_usage                                                               
 ├─ CPU allocation time (tier 2)                          Monthly cost depends on usage: $0.0000336 per vCPU-seconds   
 ├─ Memory allocation time (tier 2)                       Monthly cost depends on usage: $0.0000035 per GiB-seconds    
 └─ Number of requests                                    Monthly cost depends on usage: $0.0000004 per requests       
                                                                                                                       
 OVERALL TOTAL                                                                                              $183.67 

*Usage costs can be estimated by updating Infracost Cloud settings, see docs for other options.

──────────────────────────────────
3 cloud resources were detected:
∙ 3 were estimated

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━┓
┃ Project                                            ┃ Baseline cost ┃ Usage cost* ┃ Total cost ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━╋━━━━━━━━━━━━┫
┃ main                                               ┃          $184 ┃           - ┃       $184 ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━┻━━━━━━━━━━━━┛